hail/python/hailtop/batch/batch.py

```diff
@@ -194,10 +204,11 @@ async def _serialize_python_functions_to_input_files(
         self, path: str, dry_run: bool = False
     ) -> None:
         for function_id, function in self._python_function_defs.items():
-            file = await self._serialize_python_to_input_file(
-                path, "functions", function_id, function, dry_run
-            )
-            self._python_function_files[function_id] = file
+            if function_id not in self._python_function_files:
```

Comment:
I think you no longer need these changes. The if statement will always be true since you clear out the dictionary between submissions.